src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs

```diff
@@ -193,20 +193,20 @@ public bool IsPackageInstalled(Project project, string packageId)
 
         public bool IsPackageInstalledEx(Project project, string packageId, string versionString)
         {
-            SemanticVersion version;
+            NuGetVersion version;
             if (versionString == null)
             {
                 version = null;
             }
-            else if (!SemanticVersion.TryParse(versionString, out version))
+            else if (!NuGetVersion.TryParse(versionString, out version))
             {
-                throw new ArgumentException(VsResources.InvalidSemanticVersionString, "versionString");
+                throw new ArgumentException(VsResources.InvalidNuGetVersionString, "versionString");
             }
 
             return IsPackageInstalled(project, packageId, version);
         }
 
-        public bool IsPackageInstalled(Project project, string packageId, SemanticVersion version)
+        public bool IsPackageInstalled(Project project, string packageId, NuGetVersion version)
```

Comment:
Keep the existing method and and convert with NuGetVersion.Parse(verison.ToString())

Add a second internal method that takes a NuGetVersion. Changing this public API from SemanticVersion to NuGetVersion may cause problems even though it is more or less impossible to use.